### PR TITLE
watch for more functions inserts/updates

### PR DESCRIPTION
**Pull request by ziluvatar**
It was watching only `insert`,`insertMany` and `save`. This PR make it watch for more functions `update`, `updateOne`,`updateMany` and `insertOne`.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -191,15 +191,40 @@ class MongoWatcher extends EventEmitter {
...
+
+      const updateFunctions = [
+        'update',
+        'updateOne',
+        'updateMany'
+      ];
+
+      updateFunctions.forEach((updateFnName) => {
+        collectionInstance[updateFnName] = (function(updateFn) {
+          return function(query, document) {
```
Comment (luuuis) [inline index.js]: This is the same logic as for insert, I would consider to abstract it into a function:
- param: a collection name
- param a function that extracts the document from `arguments`
- return: decorated function

File: test/large.document.insert.tests.js [hunk 2]
```diff
@@ -62,58 +62,89 @@ describe('large.document.insert', function () {
       after(cleanCollection);
       after(removeEventListeners);
 
+      const insertCases = [
+        {
+          name: 'save',
+          insertFn: (coll, doc, cb) => coll.save(doc, cb)
```
Comment (luuuis) [inline test/large.document.insert.tests.js:68]: Should we also test the promise-returning flavour of these methods?

File: test/large.document.insert.tests.js [hunk 3]
```diff
@@ -62,58 +62,89 @@ describe('large.document.insert', function () {
...
-
-          collection.save(badDoc, () => {});
         });
       });
 
 
       describe('when object is smaller than threshold', function(){
-        it('should not emit the event when inserting', function(done) {
-          const goodDoc = buildObjectSmallerThan(testCase.threshold);
+        insertCases.concat(updateCases).forEach(function(insertCase){
```
Comment (luuuis) [inline test/large.document.insert.tests.js]: (nitpick) `insertCase` it not appropriate here

File: index.js (hunk 1 above)
Comment (ziluvatar) [reply index.js]: I'll give a look, I didn't want to play with `arguments`.

File: test/large.document.insert.tests.js (hunk 2 above)
Comment (ziluvatar) [reply test/large.document.insert.tests.js:68]: True! thanks! I forget about it.

File: index.js (hunk 1 above)
Comment (luuuis) [reply index.js]: No need to change `arguments`, right? You just need to know how to get `document` from there (it's either the 1st or 2nd arg).

Anyhow this is pretty minor. ¯\_(ツ)_/¯

Comment (ziluvatar) [reply index.js]: Take a look to see if something like that was what you were looking for (I didn't understand the interface you proposed with a a "collection name")

**Review by luuuis — APPROVED** [review]

**Comment by siacomuzzi** [comment]
`node-mongodb-watcher#v1.3.0`

**Pull request merged**